### ISSUE TITLE
I achieved the program, but now the results are not so good ,maybe I need try to  train the model using different params . thanks

### ISSUE THREAD

**Issue by chenchaohui** [EnVision-Research/Defect_Spectrum#4]

**Comment by AndysonYs** [comment]
Hi. Would like to show us some details about your project? Are you using defect gen on your dataset?

**Comment by chenchaohui** [comment]
when I use the defect gen to my own dataset, the results are abnomal. but  when I use huggingface data ,the results are nomal.

**Comment by chenchaohui** [comment]
> Hi. Would like to show us some details about your project? Are you using defect gen on your dataset?

hello, AndysonYs, When I use my own dataset ,the result is abnomal,,the generated data is very different with input  data .

**Comment by chenchaohui** [comment]
> Hi. Would like to show us some details about your project? Are you using defect gen on your dataset?

I use the labelme to genarate mask data，and then， transform it to converted Groundytruth mask ，but the train result is very strange。

**Comment by chenchaohui** [comment]
> Hi. Would like to show us some details about your project? Are you using defect gen on your dataset?

I need you help， how Can I communicate with you soon. can you give me you wechat number or qq number,thank you very much.

**Comment by chenchaohui** [comment]
> Hi. Would like to show us some details about your project? Are you using defect gen on your dataset?

Maybe there is some mistakes when I make my own  train mask, so I need you help maybe.

**Comment by chenchaohui** [comment]
> Hi. Would like to show us some details about your project? Are you using defect gen on your dataset?

I Rent cloud server with 4 RTX3090 ，However，the train speed with one RTX3090 is faster than four RTX3090？

**Comment by AndysonYs** [comment]
> > Hi. Would like to show us some details about your project? Are you using defect gen on your dataset?
> 
> hello, AndysonYs, When I use my own dataset ,the result is abnomal,,the generated data is very different with input data .

Are you using the 2 stage defect-gen (combination of large and small receptive field models) for your data? Firstly, you can try using the large-receptive field model only and validate its performance. If the 2 stage defect-gen fails but the large model works well, that means the then you should adjust the hyper-param of the switch point of our 2 models and use more large receptive field model.

**Comment by AndysonYs** [comment]
> face data ,

To adapt to your own dataset, you may change some architecture hyper-params. For example, if your data has a higher resolution, you need to add more down-sampling layers in the diffusion unet.

**Comment by AndysonYs** [comment]
> > Hi. Would like to show us some details about your project? Are you using defect gen on your dataset?
> 
> Maybe there is some mistakes when I make my own train mask, so I need you help maybe.

Could you tell me some features of your dataset? like the amount of data, the resolution, the num of defects. you can also post some examples here if available.

**Comment by AndysonYs** [comment]
> > Hi. Would like to show us some details about your project? Are you using defect gen on your dataset?
> 
> I Rent cloud server with 4 RTX3090 ，However，the train speed with one RTX3090 is faster than four RTX3090？

It seems weird to me. Did you just change the nproc_per_node? if you change it from 1 to 4 without modifying any other hyper-params, then it means you train it for 4 times long.

**Comment by chenchaohui** [comment]
> > > Hi. Would like to show us some details about your project? Are you using defect gen on your dataset?
> > 
> > 
> > I Rent cloud server with 4 RTX3090 ，However，the train speed with one RTX3090 is faster than four RTX3090？
> 
> It seems weird to me. Did you just change the nproc_per_node? if you change it from 1 to 4 without modifying any other hyper-params, then it means you train it for 4 times long.

CUDA_VISIBLE_DEVICES="0，1，2，3" \
python -m torch.distributed.launch \
--nproc_per_node=4 \  this is the params I use. any other params should I change? can you give me other parmas name?

**Comment by chenchaohui** [comment]
> you tell me some features of your dataset? like the amount of data, the resoluti
the features of my dataset as follows:
![fracture_crop_07_24bit_rgb](https://github.com/user-attachments/assets/1aa95b2b-534d-4613-b66f-67eca052574c)
![scatter_13_24bit](https://github.com/user-attachments/assets/688486f0-0580-45d9-b589-f33296ac8f18)
resolusion: original resolution is 3840*5120, I just crop the defect part as 256*256.
the results I generated :

![samples_image_000000_0_0](https://github.com/user-attachments/assets/f04400fd-e8a7-454e-83e7-50747eec85e0)
![samples_image_000000_0_1](https://github.com/user-attachments/assets/eca5dfb4-8e67-41c4-aa1e-bc8afe401497)
![samples_image_000000_1_0](https://github.com/user-attachments/assets/1371d72b-28f3-4e2f-abd7-a3b9be93b442)
![samples_image_000000_1_1](https://github.com/user-attachments/assets/fac63167-9751-4628-87ab-6c3e75128ba6)
![samples_image_000000_2_0](https://github.com/user-attachments/assets/1e8b4627-90fd-4347-97b6-7b87e2e203f2)
![samples_image_000000_2_1](https://github.com/user-attachments/assets/03c25159-6cb1-4731-a2b1-9f72dba34111)
![samples_image_000000_3_0](https://github.com/user-attachments/assets/edb08a35-289a-4c9b-ab4a-a9f3e9c045be)

**Comment by chenchaohui** [comment]
> > > Hi. Would like to show us some details about your project? Are you using defect gen on your dataset?
> > 
> > 
> > hello, AndysonYs, When I use my own dataset ,the result is abnomal,,the generated data is very different with input data .
> 
> Are you using the 2 stage defect-gen (combination of large and small receptive field models) for your data? Firstly, you can try using the large-receptive field model only and validate its performance. If the 2 stage defect-gen fails but the large model works well, that means the then you should adjust the hyper-param of the switch point of our 2 models and use more large receptive field model.

yes,I use the 2 stage defect-gen for my data,  the hyper-param of the switch point of our 2 models  mean the param --step_inference 400 ,this one?

**Comment by zhifeichen097** [comment]
> > > > Hi. Would like to show us some details about your project? Are you using defect gen on your dataset?
> > > 
> > > 
> > > hello, AndysonYs, When I use my own dataset ,the result is abnomal,,the generated data is very different with input data .
> > 
> > 
> > Are you using the 2 stage defect-gen (combination of large and small receptive field models) for your data? Firstly, you can try using the large-receptive field model only and validate its performance. If the 2 stage defect-gen fails but the large model works well, that means the then you should adjust the hyper-param of the switch point of our 2 models and use more large receptive field model.
> 
> yes,I use the 2 stage defect-gen for my data, the hyper-param of the switch point of our 2 models mean the param --step_inference 400 ,this one?

From the results you provided, it seems like the smaller model has too much involvement, which may disrupt the overall geometry of the image. I think you should start with the large receptive model only (exclude the small model, you can do that by commenting out the small model and setting the step_inference to 0, meaning you are only using the large model for inference) first. After verifying the image quality, you can start tuning the switching step by adjusting the same parameter. The switching parameter may works different in your dataset than ours.

**Comment by chenchaohui** [comment]
> > > > > Hi. Would like to show us some details about your project? Are you using defect gen on your dataset?
> > > > 
> > > > 
> > > > hello, AndysonYs, When I use my own dataset ,the result is abnomal,,the generated data is very different with input data .
> > > 
> > > 
> > > Are you using the 2 stage defect-gen (combination of large and small receptive field models) for your data? Firstly, you can try using the large-receptive field model only and validate its performance. If the 2 stage defect-gen fails but the large model works well, that means the then you should adjust the hyper-param of the switch point of our 2 models and use more large receptive field model.
> > 
> > 
> > yes,I use the 2 stage defect-gen for my data, the hyper-param of the switch point of our 2 models mean the param --step_inference 400 ,this one?
> 
> From the results you provided, it seems like the smaller model has too much involvement, which may disrupt the overall geometry of the image. I think you should start with the large receptive model only (exclude the small model, you can do that by commenting out the small model and setting the step_inference to 0, meaning you are only using the large model for inference) first. After verifying the image quality, you can start tuning the switching step by adjusting the same parameter. The switching parameter may works different in your dataset than ours.

when I just use the large reception model to infer, The result is also not so good ,just like the image I supply above.   Why

**Comment by chenchaohui** [comment]
> se the large reception model to infer, The result is also not so good ,just like the image I supply abov

I think my data type just like you data screw thread. from your paper，it shows the defect-gen result is so amazing,,but my experiment shows a bad result, now what should i do the achieve the the result like, dear author ,I need you help ,please help me ,thank you very much.

**Comment by congyao123456** [comment]
> > 你好。想向我们展示有关您项目的一些细节吗？您是否在数据集上使用缺陷生成？
> 
> 我需要你的帮助，我该如何尽快与你沟通。你能给我你的微信号或QQ号吗，非常感谢。

您好，我也需要技术帮助，请问能否给我你的微信或者qq呢
